src/Slices/ServiceInstanceDetails/UI/Tabs/EventTabContent.tsx

```diff
@@ -1,10 +1,300 @@
+import { InstanceLog } from "@/Slices/ServiceInstanceHistory/Core/Domain";
+import { DependencyContext, words } from "@/UI";
+import {
+  ErrorView,
+  EventIcon,
+  LoadingView,
+  DateWithTimeDiffTooltip,
+  Link,
+} from "@/UI/Components";
+import { InstanceDetailsContext } from "../../Core/Context";
+import { CompileReportLink } from "../Components/CompileReportLink";
+import { TabContentWrapper } from "./TabContentWrapper";
+
+interface Props {
+  selectedVersion: string;
+}
+
+/**
+ * EventsTabContent
+ *
+ * transitions and their events.
+ * The context is lsm instances changing state or trying to change state.
+ * As such,
+ * 1. each event is bound to an lsm instances
+ * An event means that some trigger has arrived to make the instance change state.
+ * As such,
+ * 2. the event is bound to version of the lsm instance. (The version in which the event was received, which is sometimes a bit confusing)
+ * 3. the event is bound to a specific type of transfer, e.g. create or update
+ * 4. it may carry information about the trasnfer (source, destination...)
+ * To make the transition, mutliple event can be logged, but
+ * 5. if multiple events they are about the same transition, they carry the same event_correlation_id
+ * When going through a resource transfer, it may start out as a failure transfer (i.e. if the lsm handler signals it to be a failure)
+ * 6. on a resource based transfer, is_error_transition is set on all the event when we are going to the error state
+ * When doing a validation compile, the transfer starts out as a normal transfer (we assume success)
+ * 7. When the validation fails however, we update the is_error_transition to fails on the next event (correlation id remains the same)
+ * Sometimes, it happens that a transfer races another one (set state while doing a validation compile e.g.)
+ * In that case, a trigger is received, but it is not executed.
+ * 8. when a transfer is received, but not executed, we log an ignored_transition
+ *
+ * The easiest way of looking at it is to consider all event with the same correlation id together.
+ * For the is_error_transition  and ignored_transition the state of the last event in the sequence it most relevant
+ *
+ * @note We are using the events that are pressent in the history logs for the instance details view.
+ * A link button allows the user to navigate the the full events page.
+ *
+ * @prop {Props} props - The EventsTabContent Props
+ *  @prop {string} selectedVersion - The actively selected version on the page.
+ *
+ */
+export const EventsTabContent: React.FC<Props> = ({ selectedVersion }) => {
+  const { logsQuery, instance } = useContext(InstanceDetailsContext);
+  const { routeManager } = useContext(DependencyContext);
+  const [expanded, setExpanded] = useState<string[]>([]);
+
+  if (logsQuery.isLoading) {
+    return (
+      <TabContentWrapper id="events">
+        <LoadingView />
+      </TabContentWrapper>
+    );
+  }
+
+  if (!logsQuery.data) {
```

Comment:
Hmm I wonder,  couldn't this view have different order

`if(logsQuery.isSuccess)` condition would have that logic about selected version
I don't see assertion for `logsQuery.isError` here, I assume it's somewhere as this query got initialised somewhere. but if not then instead of ` if (!logsQuery.data)` I would add the one for Error

and at the end we fall back to isLoading as a default return, then we don't need to assert that and then there is no need for ` if (!logsQuery.data)` condition

Reply:
No, because the data is used in the next parts of the code, so I do need the check before using it

Concerning the isError assertion, it is equivalent to !logsQuery data, because the query will fire isError when there is no data at this point. I can add the assertion if you want it

Reply:
if logsQuery.isSuccess is true then it will automatically mean that the data is defined, unless interface of the query said different(succesful query can return undefined, but I don't think it's the case here) and then I would change the assertion to the isError, it follows what react query flow and is a bit better if it comes to readability, also you current error handling shows user that there is no data, but no actual error message

Reply:
If there is data, it also automatically means the query is successful. 
Like I said, I can add an additional assertion if you want, or a comment in the code for readability, but the `!logsQuery.data` is required for further use in the code. 

What other error message do you want to display 🤔
You'll have the query error message in the History section, and one in the relevant tabs. 
You'll get:  "There is no events data available for this version."  Which matches what would be displayed on other tabs and  "Error loading Version History" in the History section on the right side of the screen.